test/e2e-test-application/cypress/e2e/tests/1-angular/luigi-client-link-manager-features-4.cy.js

```diff
@@ -227,110 +192,56 @@ describe('Luigi Client linkManager Webcomponent, Drawer', () => {
     });
 
     it('open webcomponent with visibleForFeatureToggles', () => {
-      cy.window().then(win => {
-        cy.wait(500);
-        cy.get('.wcContainer>div>div>*').then(container => {
-          const root = container.children().prevObject[0].shadowRoot;
-          const wcContent = root.querySelector('p').innerText;
-          expect(wcContent).to.equal('Some input text !ft');
-        });
-      });
+      cy.get('.wcContainer>div>div>[nodeid="input1"]')
+        .shadow()
+        .contains('p', 'Some input text !ft')
+        .should('exist');
     });
   });
 
   describe('Webcomponent compound view test', () => {
     beforeEach(() => {
       cy.visitLoggedIn('/projects/pr1/wc_grid_compound');
+      cy.get('.wcContainer>div>div>[nodeid="btn"]')
+        .shadow()
+        .find('button')
+        .as('btn');
+      cy.get('.wcContainer>div>div>[nodeid="timer"]')
+        .shadow()
+        .find('p')
+        .as('timer');
     });
 
     it('open webcomponent btn', () => {
-      cy.window().then(win => {
-        cy.wait(700);
-        cy.get('.wcContainer>div>div>*').then(container => {
-          cy.wait(500);
-          const root = container.children().prevObject[0].shadowRoot;
-          const wcContent = root.querySelector('button').innerText;
-
-          expect(wcContent).to.equal('Start');
-        });
-      });
+      cy.get('@btn').should('have.text', 'Start');
```

Comment:
Just as a question. What is this `@btn` notation. I haven't seen it before 
Also below `@timer ` below 🤔 . Seems to be quite a shorter solution indeed.

Reply:
Cypress calls those aliases. When used with HTML elements, this lets you save and replay queries:
https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Elements

Useful stuff and also makes it easier to use `.should()` instead of `.then()`. At least in a few cases.

The aliases used here are created in the `beforeEach` hook.